FuelLabs/sway
# Transition to tracing in forc (and throughout sway) over println

closes #884.

### Possible testing steps

Since this PR touches a lot of files (with little adjustments), I am providing testing steps and outputs that cover all changed files. I tried to supply raw text outputs as much as possible but in some places, photos are used to preserve colors as well. 

- [ ]  forc:
<details>
<summary>forc - testing steps</summary>

-  forc init - success:

    1. Run `cargo run --bin forc --init example_prog`

  - Output before `tracing`:
 
    <img width="525" alt="image" src="https://user-images.githubusercontent.com/20915464/169046233-b2b7123d-b650-41df-b524-e20c1229d82c.png">


  - Output after `tracing`:
    <img width="525" alt="image" src="https://user-images.githubusercontent.com/20915464/169046083-3de4c03a-15b8-414b-835f-23327598390c.png">
  
--------------------------------------------
-  forc init - error:

    1. Run `cargo run --bin forc -- init --template error example_prog`

  - Output before `tracing`:
    ```
    Error: Failed to initialize project from a template with the given name "error": https://api.github.com/repos/FuelLabs/sway/contents/examples/error: status code 404.
      Note: If you are attempting to initialize this project from a Sway example, please ensure the template name matches one of the available examples.
	- asm_return_tuple_pointer
	- counter
	- fizzbuzz
	- hashing
	- liquidity_pool
	- match_statements
	- msg_sender
	- native_token
	- signatures
	- storage_example
	- subcurrency
	- wallet_smart_contract


    ```
  - Output after `tracing`:

    ```
    Error: Failed to initialize project from a template with the given name "error": https://api.github.com/repos/FuelLabs/sway/contents/examples/error: status code 404.
      Note: If you are attempting to initialize this project from a Sway example, please ensure the template name matches one of the available examples.
	- asm_return_tuple_pointer
	- counter
	- fizzbuzz
	- hashing
	- liquidity_pool
	- match_statements
	- msg_sender
	- native_token
	- signatures
	- storage_example
	- subcurrency
	- wallet_smart_contract


    ```
</details>

- [ ]  forc-fmt:
<details>
<summary>forc-fmt - testing steps</summary>

-  forc-fmt in the root directory (outside of a sway project):

    1. Run `cargo run --bin forc-fmt`

  - Output before `tracing`:
    ```
    Error: Manifest file does not exist
    ```
  - Output after `tracing`:

    ```
    Error: Manifest file does not exist
    ```
--------------------------------------------
-  forc-fmt in an example sway project directory:

    1. Run `cargo run --bin forc -- init example_prog`
    2. Run `cargo run --bin forc-fmt -- --path example_prog`
    3. Run `cargo run --bin forc-fmt -- --path example_prog --check`

  - Output before `tracing`:
    ```
    Manifest Forc.toml properly formatted
    ```
  - Output after `tracing`:

    ```
    Manifest Forc.toml properly formatted
    ```
</details>

- [ ]  forc-explore:
<details>
<summary>forc-explore - testing steps</summary>

-  forc-explore - success

    1. Run `cargo run --bin forc-explore`

  - Output before `tracing`:
    ```
    Fuel Network Explorer 0.1.1
    Running server on http://127.0.0.1:3030
    ```
  - Output after `tracing`:

    ```
    Fuel Network Explorer 0.1.1
    Running server on http://127.0.0.1:3030
    Server::run{addr=127.0.0.1:3030}: listening on http://127.0.0.1:3030
    ```
--------------------------------------------
-  forc-explore - error

    1. Run `cargo run --bin forc-explore` on a second terminal
    2. Run `cargo run --bin forc-explore` on the main terminal


  - Output before `tracing`:
    ```
    Fuel Network Explorer 0.1.1
    Running server on http://127.0.0.1:3030
    thread 'main' panicked at 'error binding to 127.0.0.1:3030: error creating server listener: Address already in use (os error 48)', /Users/kayagokalp/.cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.2/src/server.rs:213:27
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    ```
  - Output after `tracing`:

    ```
    Fuel Network Explorer 0.1.1
    Running server on http://127.0.0.1:3030
    thread 'main' panicked at 'error binding to 127.0.0.1:3030: error creating server listener: Address already in use (os error 48)', /Users/kayagokalp/.cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.3.2/src/server.rs:213:27
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    ```
</details>
